cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -279,12 +286,12 @@ public ApplicationDetail getAppDetail(ApplicationId appId) throws Exception {
    * @throws Exception if ensureAccess throws exception
    */
   public boolean updateAppAllowed(ApplicationId appId) throws Exception {
-    ensureAccess(appId);
     ApplicationSpecification appSpec = store.getApplication(appId);
     if (appSpec == null) {
       // App does not exist. Allow to create a new one
       return true;
     }
+    ensureAccess(appId);
```

Comment:
This is to make sure we do not expose the existence of the app, and this method is actually not doing the update, it is checking whether this app can be deployed with the app version provided, maybe we should change the name of this method